src/Adyen/HttpClient/CurlClient.php

```diff
@@ -155,7 +155,7 @@ public function requestPost(\Adyen\Service $service, $requestUrl, $params)
 
         if (!in_array($httpStatus, $resultOKHttpStatusCodes) && $result) {
```

Comment:
This code block is repeated 3 times in this class. can we generalise this to reduce code duplication?